google-http-client/src/main/java/com/google/api/client/http/HttpResponseException.java

```diff
@@ -185,7 +210,19 @@ public Builder(HttpResponse response) {
       // message
       StringBuilder builder = computeMessageBuffer(response);
       if (content != null) {
-        builder.append(StringUtils.LINE_SEPARATOR).append(content);
+        try {
+          // charset
+          charset = response.getContentCharset();
```

Comment:
`getContentCharset()` doesn't throw an exception.  Please move it up into the ` if (inputStream != null)` section.

Reply:
It can throw as it uses from Charset.java 
public static Charset forName(String charsetName) {
        Charset cs = lookup(charsetName);
        if (cs != null)
            return cs;
        throw new UnsupportedCharsetException(charsetName);
    }